src/main/java/io/reactivex/internal/operators/observable/ObservableDematerialize.java

```diff
@@ -18,9 +18,9 @@
 import io.reactivex.internal.disposables.DisposableHelper;
 import io.reactivex.plugins.RxJavaPlugins;
 
-public final class ObservableDematerialize<T> extends ObservableSource<Try<Optional<T>>, T> {
+public final class ObservableDematerialize<T> extends ObservableSource {
```

Comment:
Looks like there already was an `ObservableSource` class that we're now clashing with. I'm tempted to just inline it since all it does is hold a value in a field. As an actual type in the hierarchy it's not used in any special way. Either that or just rename it `ObservableFromUpstream` for now and delete it later.